### DISCUSSION TITLE
EACCES Permission Denied Error on LibreChat Startup on Debian 12  What happened?

### DISCUSSION THREAD

**Discussion by miztizm** [danny-avila/LibreChat#2141] [Troubleshooting]
### What happened?

When attempting to start the LibreChat application on Debian 12, the process exits with code 1 after throwing an EACCES (permission denied) error while trying to open a log file (/app/api/logs/error-2024-03-19.log). Additionally, a message indicating that Redis has not been initialized is logged, noting that Redis support is experimental. This issue prevents the LibreChat application from starting successfully.

### Steps to Reproduce

- Ensure the system is running Debian 12.
- Install the LibreChat application as per the documentation.
- Start the application using docker-compose up
- Observe the error output and application exit.

### Relevant log output

```shell
chat-mongodb   | {"t":{"$date":"2024-03-19T20:01:28.230+00:00"},"s":"I",  "c":"CONTROL",  "id":23138,   "ctx":"initandlisten","msg":"Shutting down","attr":{"exitCode":100}}
LibreChat      |
LibreChat      | > LibreChat@0.6.10 backend
LibreChat      | > cross-env NODE_ENV=production node api/server/index.js
LibreChat      |
LibreChat      | 2024-03-19 20:01:29 info: [Optional] Redis not initialized. Note: Redis support is experimental.
LibreChat      | 2024-03-19 20:01:30 error: There was an uncaught error: EACCES: permission denied, open '/app/api/logs/error-2024-03-19.log'
LibreChat exited with code 1
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by fuegovic** [comment]
try manually creating the required directories and give them the necessary permissions:
example:
```bash
# Create necessary directories
mkdir -p /app/uploads/temp
mkdir -p /app/client/public/images/temp
mkdir -p /app/api/logs/
mkdir -p /app/data

# Give write permission to the directory
chmod -R 777 /app/uploads/temp
chmod -R 777 /app/client/public/images
chmod -R 777 /app/api/logs/
chmod -R 777 /app/data
```

**Comment by 0x7466** [comment]
Same problem here!

**Reply by danny-avila** [reply]
Grant permissions, as shown in comment above https://github.com/danny-avila/LibreChat/discussions/2141#discussioncomment-8845459, or build the docker image yourself, see here related: https://github.com/danny-avila/LibreChat/discussions/1441

**Reply by 0x7466** [reply]
Thanks, I'll have a look.

My problem is that the container is crashing before I'm able to change any permission. I've also tried to delete the logs but it looks like it still tries to access them.

Awesome project btw!!

**Reply by danny-avila** [reply]
Thank you!

Make sure you're on the latest commit. You may also need to edit the dockerfile as shown here: https://github.com/danny-avila/LibreChat/discussions/1441#discussioncomment-7959185

You can make use of the override file to build the image locally, did you have issues with this?

https://docs.librechat.ai/install/configuration/docker_override.html#step-2-edit-the-override-file
